src/connect4/ai.rs

```diff
@@ -2,196 +2,252 @@
 
-    fn find_win_probability(&self, board: GameState, currMove: i32, lastMove: i32) -> f32 {
+    fn find_win_probability(&self, board: Board, currMove: i32, lastMove: i32) -> f32 {
         //Check win for AI turn
-        let moves = Vec::new();
-        for i in 0..core::BOARD_SIZE.0 {
-            if currMove & 2 == 0 {
-                let oppMove = MoveCheck::init(board, i, self.team%2 + 1);
-                //If move would cause other team to win, return 0 as prob. Otherwise, call recursively on next move
-                if oppMove.has_end_result() {
-                    return 0;
+        let mut moves = Vec::new();
+        let mut out = "".to_owned();
+        for i in 0..BOARD_SIZE.1 {
+            if !board.is_column_full(i as usize) {
+                out += "non-full column\n";
+                let board = board.clone();
+                //This will always make a MoveCheck where the "team" is self.team if currMove%2 == 0 and the opposite team if currMove%2 == 1
+                //Assumes only two teams, 1 and 2
+                let moveCheck = MoveCheck::new(board, i, (self.team+currMove+1)%2+1);
+                //If move produces end result, return an absolute probability of 1 (if current moveCheck is for team) or 0 (for opp)
+                if moveCheck.has_end_result() {
+                    return (1-(self.team-moveCheck.team).abs()%2) as f32;
+                //If currMove is not last move, recurse on subsequent moves from the current moveCheck
                 } else if currMove < lastMove {
-                    return self.find_win_probability(oppMove.board, currMove+1, lastMove)
+                    moves.push(self.find_win_probability(moveCheck.board, currMove+1, lastMove));
+                    out += "pushed move\n";
+                //Base case - this is the last move, so just return current probability of win for this moveCheck relative to self.team
                 } else {
-                    moves.push(oppMove);
+                    out += "finised move\n";
+                    moves.push(moveCheck.get_win_probability(self.team));
                 }
-            } else {
-                let aiMove = MoveCheck::init(board, i, self.team);
-                //If move would cause AI to win, return 1 as prob. Otherwise, call recursively on next move
-                if aiMove.has_end_result() {
-                    return 1;
-                } else if currMove < lastMove {
-                    return self.find_win_probability(aiMove.board, currMove+1, lastMove);
-                } else {
-                    moves.push(oppMove);
+            }
+        }
+        //Edge case - no moves to make, return 0 probability (can't win)
+        if moves.len() == 0 {
+            0f32
+        //Otherwise, return average of all possibilities
+        } else {
+            moves.iter().sum::<f32>()/(moves.len() as f32)
+        }
+    }
+}
+
+
+#[cfg(test)]
+mod ai_tests {
+    use super::*;
+    use connect4::core::Board;
```

Comment:
`use` was weird...I used `super::*` to get access to things in the outer scope (i.e. things in `AI`, but sometimes you still needed to import things like `Board` even though I had `use connect4::core::Board` in the file scope as well.